### DISCUSSION TITLE
Thoughts on this approach to Multi-SKU model?

### DISCUSSION THREAD

**Discussion by allisonwiener** [google/lightweight_mmm#263] [Q&A] [answered]
LightweightMMM is unable to run multiple SKUs within one model despite being able to handle multi geographies. I've been gathering workarounds to this as it's a requirement for my models to be run in one vs many & would love this communities thoughts on the following 2 approaches. Any watch-outs? Do you think this will work? Thank you!

**Method 1:** Feeding multiple brands/channels into the model as if they were just many media channels. Then you can layer multiple geographies on top and run multi geo/SKUs in one model. Example here: https://forecastegy.com/posts/how-to-create-a-marketing-mix-model-with-lightweightmmm/ 

**Method 2:** Running multiple SKUs through the Geo Level (regional) approach by changing the data from regions to SKUs to reflect each item as the cross-section rather than the traditional regional identifier. I won't be able to easily include effects that occur between the SKUs (substitution effects), but is it correct to assume results should more or less be correct if these products don't tend to have much substitution amongst themselves?

**Comment by DiddyDaw** [comment, marked as answer]
Caveat to the below is i know MMM very well (have been building models for 25 years) but am not an expert on the intricate workings of the Lightweight implementation;

What I see as being difficult in either method is judging the inter-relationships between the SKUs as you rightly highlight in method 2.

If they are independent SKUs then each should be modelled independently but that seems to go counter to your requirements. Is it really not possible to write a loop around lightweightmmm or process models in parallel then bring the results together at the end?

What I would worry about with lightweight MMM is how marketing resources are allocated to different SKUs. For example, do you have some advertising that targets individual SKUs whilst other campaigns target multiple SKUs? This kind of hierarchical campaign structure is taken care of in specialist packages but i can't see how lightweightmmm can take account of it.

Method 1 doesn't make sennse to me at all. How would you organise the dependent variable to account for all the SKUs? You can't stack them as any carryover effects would blend between SKUs.

If it were me, i would try the loop approach across multiple SKUs with a procedure to make sure that halo / cannibalisation variables are also created within the data sets. I would then create a results procedure at the end to unify the results into one final table - thus giving you all the lifts in the same place.
